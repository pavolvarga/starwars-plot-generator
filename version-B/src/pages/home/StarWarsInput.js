import React, { Component } from 'react';
import {Col, FormGroup, Input, Label} from "reactstrap";

class StarWarsInput extends Component {

    constructor(props) {
        super(props);
        this.state = {
            selectedFromData: false
        };
        this.handleOnChange = this.handleOnChange.bind(this);
    }

    handleOnChange(value, data, setFn) {

        //ignore empty or too short text
        if (!value || value.length < 3) {
            this.setState(() => {
                return {selectedFromData: false}
            });
            setFn(undefined);
            return;
        }

        const foundIdx = data.findIndex(el => el.name === value);
        if (foundIdx !== -1) {
            this.setState(() => {
                return {selectedFromData: true}
            });
            setFn(data[foundIdx]);
        }
        else {
            this.setState(() => {
                return {selectedFromData: false}
            });
            setFn(undefined);
        }
    }

    render() {
        const {id, name, label, placeholder, data, setFn, visible} = this.props;

        if (!visible) {
            return null;
        }

        return (
            <Col>
                <FormGroup size="lg" row>
                    <Label for="input-person" size="lg">{label}</Label>
                    <Input
                        id={id}
                        name={name}
                        placeholder={placeholder}
                        bsSize="lg"
                        valid={this.state.selectedFromData ? true : undefined}
                        onChange={e => this.handleOnChange(e.target.value, data, setFn)}
                        disabled={data.length === 0} />
                </FormGroup>
            </Col>
        );
    }
}

export { StarWarsInput };