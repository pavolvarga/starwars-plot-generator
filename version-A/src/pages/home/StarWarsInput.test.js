import React from 'react';
import { mount } from 'enzyme/build';
import { StarWarsInput } from "./StarWarsInput";

const NOOP = () => {};

describe('StarWarsInput', () => {
    function createStarWarsInput() {
        return mount(<StarWarsInput id={''} name={''} placeholder={''} data={[]} setFn={NOOP} visible={true}/>);
    }
    it('should render correctly in "debug" mode', () => {
        const component = createStarWarsInput();
        expect(component).toMatchSnapshot();
    });
});
