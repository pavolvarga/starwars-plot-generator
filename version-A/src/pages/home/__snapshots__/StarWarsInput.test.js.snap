// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarWarsInput should render correctly in "debug" mode 1`] = `
<StarWarsInput
  data={Array []}
  id=""
  name=""
  placeholder=""
  setFn={[Function]}
  visible={true}
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <div
      className="col"
    >
      <FormGroup
        row={true}
        size="lg"
        tag="div"
      >
        <div
          className="row form-group"
          size="lg"
        >
          <Label
            for="input-person"
            size="lg"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <label
              className="col-form-label-lg"
              htmlFor="input-person"
            />
          </Label>
          <Input
            bsSize="lg"
            disabled={true}
            id=""
            name=""
            onChange={[Function]}
            placeholder=""
            type="text"
          >
            <input
              className="form-control-lg form-control"
              disabled={true}
              id=""
              name=""
              onChange={[Function]}
              placeholder=""
              type="text"
            />
          </Input>
        </div>
      </FormGroup>
    </div>
  </Col>
</StarWarsInput>
`;
